docs/design/2023-09-21-tiflash-resource-control.md

```diff
@@ -0,0 +1,72 @@
+# TiFlash Resource Control
+## Introduction
+This document introduces the design and implementation of TiFlash's resource management feature.
+
+## Motivation and Background
+TiDB currently supports global resource management to support multiple businesses in a cluster and achieve isolation between them.
+
+As part of the TiDB architecture, TiFlash also needs to support resource management to achieve the capability of isolating different businesses, especially in AP/HTAP scenarios.
+
+## Detailed Design
+The basic idea is similar to TiDB/TiKV's resource control. It adopts a combination of rate limiting and priority scheduling:
+1. Rate Limiting: When scheduling tasks, it limits them based on the Token Bucket algorithm.
+2. Priority: When physical resources are scarce, tasks are scheduled based on their priority, determining which task should run first.
+
+The underlying idea behind this design is as follows: If the `RU_PER_SEC` setting for each resource group is relatively small, and the sum of these settings does not exceed the real physical resource limit, rate limiting plays the primary role in throttling each resource group. However, to ensure that resources are used more efficiently and to prevent situations where a small `RU_PER_SEC` setting leads to unused physical resources, TiDB's resource management mechanism allows oversubscription, meaning that the total `RU_PER_SEC` values can exceed the physical resources. In such cases, priority scheduling comes into play to prevent resource contention among resource groups.
```

Comment:
What does "relatively small" mean here?

Reply:
1 RU means 3ms cpu or 64KiB read bytes. If `RU_PER_SEC` is `relatively small`, it means the physical resources are  sufficient for all currenty queries to use.

Reply:
Does it mean "the sum of these settings does not exceed the real physical resource limit"? If so, I'd suggest remove the "relatively small" statement.

Reply:
how about:
```
If the `RU_PER_SEC` setting for each resource group is relatively small(a.k.a. the sum of these settings does not exceed the real physical resource limit), then rate limiting plays the primary role in throttling each resource group.
```

Reply:
done